docs/ilm-how-to.asciidoc

```diff
@@ -1,18 +1,191 @@
+|30 days / 50 gb
+|10 days
+
+|`metrics-apm.internal`
+|30 days / 50 gb
+|90 days
+
+|`metrics-apm.app`
+|30 days / 50 gb
+|90 days
+
+|`logs-apm.error`
+|30 days / 50 gb
+|10 days
+
+|===
+
+The APM index lifecycle policies can be viewed in {kib}.
+Navigate to *Stack Management* / *Index Lifecycle Management*, and search for `apm`.
+
+[discrete]
+[[data-streams-custom-policy]]
+=== Configure a custom index lifecycle policy
+
+This tutorial explains how to apply a custom index lifecycle policy to the
+`traces-apm` data stream with the `default` namespace.
+
+[discrete]
+[[data-streams-custom-one]]
+=== Step 1: View data streams
+
+The **Data Streams** view in {kib} shows you the data streams,
+index templates, and index lifecycle policies associated with a given integration.
+
+. Navigate to **Stack Management** > **Index Management** > **Data Streams**.
+. Search for `apm` to see all data streams associated with APM data.
+. Select the `traces-apm-default` data stream to view its associated index template and ILM policy.
+As you can see, the data stream follows the <<apm-data-streams-naming-scheme>> and starts with its type, `traces-`.
++
+[role="screenshot"]
+image::images/data-stream-overview.png[Data streams info]
+
+[discrete]
+[[data-streams-custom-two]]
+=== Step 2: Create an index lifecycle policy
+
+. Navigate to **Stack Management** > **Index Lifecycle Policies**.
+. Click **Create policy**.
+
+Name your new policy. I've chosen `custom-traces-apm-default-policy`.
+Customize the policy to your liking, and when you're done, click **Save policy**.
+
+[discrete]
+[[data-streams-custom-three]]
+=== Step 3: Create a component template
+
+To apply your new index lifecycle policy to a data stream,
+and to ensure the policy continues to be applied in future versions,
+you must create a component template.
+The component template name must start with `.`, follow the data stream naming scheme,
+and end with `@custom`:
+
+[source,text]
+----
+.<type>-<dataset>-<namespace>@custom
+----
+
+For example, to create custom index settings for the `traces-apm` data stream with a namespace of `default`,
+the component template name would be:
+
+[source,text]
+----
+.traces-apm-default@custom
+----
+
+. Navigate to **Stack Management** > **Index Management** > **Component Templates**
+. Click **Create component template**.
+. Use the template above to set the name--in this case, `.traces-apm-default@custom`. Click **Next**.
+. Under **Index settings**, set the ILM policy name created in the previous step:
++
+[source,json]
+----
+{
+  "lifecycle": {
+    "name": "custom-traces-apm-default-policy"
+  }
+}
+----
+. Continue to **Review** and ensure your request looks similar to the image below.
+If it does, click **Create component template**.
++
+[role="screenshot"]
+image::images/create-component-template.png[Create component template]
+
+[discrete]
+[[data-streams-custom-four]]
+=== Step 4: Clone and modify the existing index template
```

Comment:
You shouldn't need to touch any index templates for customising the ILM policy. Just edit the `@custom` component template, which is used by the Fleet-installed index template.

Did you try that and it didn't work?

Reply:
No, sorry, I didn't try that. I was basing this off of what @hop-dev wrote for other integrations. I can try that too.

Reply:
I think you're right, @axw. Looking back at the original issue, Mark pointed out that integrations that define their own ILM policy can just edit the `@custom` template: https://github.com/elastic/observability-docs/issues/986#issuecomment-914211466

Reply:
I've confirmed I can directly edit the `@custom` component template. But that brings up two thoughts:

1. Is this persisted on upgrade?
2. I don't think this method allows the same level of granularity that creating your own index template does. For example, when I create my own index template, I can add an ILM policy that applies only to a specific namespace, like `traces-apm-production*`. With this method, my custom template is only applied to the default index pattern (all namespaces) of `traces-apm-*`. Maybe that's okay though? IDK. That's my question for you 😄

Reply:
> Is this persisted on upgrade? (not sure how to test this)

Yes

> I don't think this method allows the same level of granularity that creating your own index template does. For example, when I create my own index template, I can add an ILM policy that applies only to a specific namespace, like traces-apm-production*. With this method, my custom template is only applied to the default index pattern (all namespaces) of traces-apm-*. Maybe that's okay though? IDK. That's my question for you smile

Indeed, this is a good point. I'd separate it into use cases, so we don't lead users through unnecessary complexity when they want to do something simple. The basic use case is when you want to customise the ILM policy for a data type (e.g. traces); this can be done just be editing the `@custom` component template. The advanced use case is to customise the ILM policy for a data type and namespace (e.g. production); this requires introducing a new index template with higher priority.

Probably a good idea to document both, but longer term we need to make all of this less manual.